src/OrchardCore/OrchardCore.Testing/Infrastructure/OrchardCoreTestFixture.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Hosting;
+
+namespace OrchardCore.Testing.Infrastructure;
+
+public class OrchardCoreTestFixture<TStartup> : WebApplicationFactory<TStartup> where TStartup : class
+{
+    protected override void ConfigureWebHost(IWebHostBuilder builder)
+    {
+        var shellsApplicationDataPath = Path.Combine(Directory.GetCurrentDirectory(), "App_Data");
+
+        if (Directory.Exists(shellsApplicationDataPath))
+        {
+            Directory.Delete(shellsApplicationDataPath, true);
+        }
+
+        builder.UseContentRoot(Directory.GetCurrentDirectory());
+    }
+
+    protected override IWebHostBuilder CreateWebHostBuilder()
+        => WebHostBuilderFactory.CreateFromAssemblyEntryPoint(typeof(Program).Assembly, Array.Empty<string>());
```

Comment:
=> WebHostBuilderFactory.CreateFromAssemblyEntryPoint(typeof(TStartup).Assembly, Array.Empty<string>());
Use the assembly of the TStartup instead of reference to Program.cs in OrchardCore.Cms.Web